argument binder documentation created

meyerinj opened the pull request:
Intended to close #3425 .

Like Cindy's work on this in #3831 I am interested in input on this as I am interested in input.

graemerocher committed:
Set version to 2.2.x

pgressa committed:
Rename micronaut-discovery-client module group (#4243)

* Rename micronaut-discovery-client module group

* Fix checkstyle

micronaut-build committed:
Bump micronaut-aws to 2.2.0 (#4281)

jameskleeh committed:
Merge branch '2.1.x' into 2.2.x

micronaut-build committed:
Bump micronaut-aws to 2.2.1 (#4322)

micronaut-build committed:
Bump micronaut-grpc to 2.1.0 (#4324)

jameskleeh committed:
Merge branch '2.1.x' into 2.2.x

micronaut-build committed:
Bump micronaut-flyway to 3.0.0 (#4351)

micronaut-build committed:
Bump micronaut-sql to 3.2.0 (#4361)

meyerinj committed:
argument binder documentation created

jameskleeh commented on src/main/docs/guide/httpServer/customArgumentBinding.adoc:
```diff
@@ -0,0 +1,60 @@
+- An interface for classes that bind an api:core.type.Argument[Argument] from an api:http.HttpRequest[HttpRequest] driven by
+the argument's type.
+
+Binders are called by the api:io.micronaut.http.annotation.Controller[Controller] once the request is received and the argument
```
This paragraph is confusing to me. The filters aren't called by controllers. Both controllers and the binders are called by the server logic. I think the next paragraph is a great description, however I think an example should be given (the filters example). I would move the generic descriptions of argument binders above the specific information for typed and annotated otherwise it seems the generic information is specific to typed. The code samples need to be snippets and available in all languages and need to have tests to verify the code actually works.

meyerinj replied:
If I understood your recommendations correctly this new push should do what you suggested. Fun fact: I had never written Kotlin before this!

micronaut-build committed:
Bump micronaut-test to 2.2.0 (#4384)

micronaut-build committed:
Bump micronaut-aws to 2.2.2 (#4388)

micronaut-build committed:
Bump micronaut-test to 2.2.1 (#4399)

meyerinj committed:
Merge remote-tracking branch 'origin/2.1.x' into 3425_argument_binder_documentation

meyerinj committed:
reworked phrasing, added tests, and expanded script to include snippets

meyerinj committed:
Tests failing in Github. Removing use of java Sets in attempt to fix

jameskleeh committed:
Merge branch '2.1.x' into 2.2.x

dstepanov committed:
Reuse executable methods (#4204)

rdesgroppes committed:
Allow retrieving array element types (#4424) (#4424)

Follow-up of micronaut-projects/micronaut-openapi#368.

This is to ease post-processing without adhoc helpers such as:
https://github.com/micronaut-projects/micronaut-openapi/blob/a4524550d45c36d78061ab392bdf41d274188cf6/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java#L1553

It consists in providing a `fromArray` method to all "final" classes for
which `toArray` has to be specialized:
* `PrimitiveElement`
* `GroovyClassElement`
  * `GroovyEnumElement`
* `JavaVoidElement` (deprecated)
* `JavaClassElement`
  * `JavaEnumElement`
    (the latter was missing a specialization of `toArray` and therefore
    used to return a `JavaClassElement`, which is now fixed)

alvarosanchez committed:
New parent POM (#4433)

ilopmar committed:
Merge branch '2.1.x' into 2.2.x

micronaut-build committed:
Bump micronaut-maven-plugin to 1.1.0.M1 (#4439)

micronaut-build committed:
Bump micronaut-rss to 2.2.0 (#4437)

ilopmar committed:
Fix snapshot version

ilopmar committed:
Merge branch '2.1.x' into 2.2.x

alvarosanchez committed:
Include `org.graalvm.sdk:graal-sdk` in the BOM (#4445)

meyerinj committed:
Merge remote-tracking branch 'origin/2.2.x' into 3425_argument_binder_documentation

ilopmar committed:
Enable GraalMV `--install-exit-handlers` (#4459)

Fixes #3880

dstepanov committed:
Fix introspecting external @Entity classes (#4157)

micronaut-build committed:
Bump micronaut-data to 2.2.0 (#4462)

wetted committed:
Issue 3582 configure Netty shutdown quiet period and shutdown timeout (#4319)

* Issue 3582 - adding feature to configure the shutdown quiet period and shutdown timeout for Netty event loop groups and http server

* Issue 3582 - adding feature to configure the shutdown quiet period and shutdown timeout for Netty event loop groups and http server

* Issue 3582
- adding feature to configure the shutdown quiet period and shutdown timeouts
- adding overloaded versions of HttpClient create; RxStreamingHttpClient create; and RxHttpClientFactory createClient and createStreamingClient to accept a HttpClientConfiguration instance, for cases where the client is created outside the Micronaut application context

* Issue 3582
- adding feature to configure the shutdown quiet period and shutdown timeouts
- adding overloaded versions of HttpClient create; RxStreamingHttpClient create; and RxHttpClientFactory createClient and createStreamingClient to accept a HttpClientConfiguration instance, for cases where the client is created outside the Micronaut application context

* Issue 3582
- revise section 6.27.1 of the documentaion to add clarification about Netty server configuration
- add deprecation warning in NettyHttpServer about preferring Netty event loop groups configurations

* Update threadPools.adoc

* Issue 3582
- adding feature to configure the shutdown quiet period and shutdown timeouts
- adding overloaded versions of HttpClient create; RxStreamingHttpClient create; and RxHttpClientFactory createClient and createStreamingClient to accept a HttpClientConfiguration instance, for cases where the client is created outside the Micronaut application context

Co-authored-by: James Kleeh <james.kleeh@gmail.com>

graemerocher committed:
Cleanup oracle driver versions (#4467)

xpmatteo committed:
Enable changing logger levels from properties when using Log4j (#4286)

* Enable changing logger levels from properties when using Log4j

* Fix checkStyle in Log4jLoggingSystem

* Enable the loggers endpoint with the log4j2 feature

* Add @since 2.2.0 to Log4jLoggingSystem implementations

micronaut-build committed:
Bump micronaut-openapi to 2.2.0 (#4474)

graemerocher committed:
Create 2.3.x branch

meyerinj committed:
Merge remote-tracking branch 'origin/2.3.x' into 3425_argument_binder_documentation

meyerinj committed:
updated to run in java 8
